Infrastructure/Factories/TribalSpreadsheetTuitionPartnerFactory.cs

```diff
@@ -0,0 +1,542 @@
+            { "Organisation_Town_s", new ImportMap() { IsStoredInNtp = true, IsRequired = true, RecommendedMaxStringLength = 100 } },
+            { "Organisation_County_s", new ImportMap() { IsStoredInNtp = true, IsRequired = true, RecommendedMaxStringLength = 100 } },
+            { "Organisation_PostCode_s", new ImportMap() { IsStoredInNtp = true, IsRequired = true, RecommendedMaxStringLength = 8 } },
+            { "Organisation_Tel_s", new ImportMap(tpMapping, nameof(tpMapping.PhoneNumber)) { RecommendedMaxStringLength = 40 } },
+            { "Organisation_TP_Link_s", new ImportMap(tpMapping, nameof(tpMapping.Website)) { RecommendedMaxStringLength = 200 } },
+            { "Organisation_Email_s", new ImportMap(tpMapping, nameof(tpMapping.Email)) { RecommendedMaxStringLength = 100 } },
+            { "Organisation_Introduction_s", new ImportMap(tpMapping, nameof(tpMapping.Description)) { RecommendedMaxStringLength = 350 } },
+            { "Organisation_LegalStatus_s", new ImportMap() { IsStoredInNtp = true, IsRequired = true } },
+            { "Organisation_LogoVector_s", new ImportMap() },
+            { "Organisation_ChargeVAT_s", new ImportMap(tpMapping, nameof(tpMapping.IsVatCharged)) {IsRequired = false} },
+            { "Organisation_LastUpdated_d", new ImportMap(tpMapping, nameof(tpMapping.TPLastUpdatedData)) },
+        };
+    }
+
+    public TuitionPartner GetTuitionPartner(ISpreadsheetExtractor spreadsheetExtractor, string filename,
+        IList<Region> regions, IList<Subject> subjects, IList<OrganisationType> organisationTypes)
+    {
+        _regions = regions;
+        _subjects = subjects;
+        _organisationTypes = organisationTypes;
+
+        _warnings = new();
+        _errors = new();
+        TuitionPartner tuitionPartner = new();
+
+        _spreadsheetExtractor = spreadsheetExtractor;
+
+        if (!_spreadsheetExtractor.WorksheetExists(OrganisationDetailsSheetName))
+        {
+            _errors.Add($"Missing '{OrganisationDetailsSheetName}' worksheet");
+        }
+        if (!_spreadsheetExtractor.WorksheetExists(PricingSheetName))
+        {
+            _errors.Add($"Missing '{PricingSheetName}' worksheet");
+        }
+        if (!_spreadsheetExtractor.WorksheetExists(DeliverySheetName))
+        {
+            _errors.Add($"Missing '{DeliverySheetName}' worksheet");
+        }
+
+        if (_errors.Count == 0)
+        {
+            tuitionPartner = GetOrganisationDetails();
+
+            PopulateLocalAuthorityDistrictCoverage(tuitionPartner);
+
+            PopulateSubjectCoverageAndPrice(tuitionPartner);
+        }
+
+        if (_warnings.Any())
+        {
+            _logger.LogWarning("Issues importing Tribal spreadsheet '{filename}': {warnings}", filename, string.Join(Environment.NewLine, _warnings));
+        }
+
+        if (_errors.Any())
+        {
+            throw new Exception($"Error importing Tribal spreadsheet '{filename}': {string.Join(Environment.NewLine, _errors)}");
```

Comment:
`System.Exception` should not be thrown by user code. We should be throwing a bit more specific Exception. i.e `InvalidOperationException`.

Reply:
Agreed, change made.